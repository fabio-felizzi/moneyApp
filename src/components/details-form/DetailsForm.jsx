import React from 'react';
import PropTypes from 'prop-types';

const DetailsForm = ({
    className, formDetails, submitForm, submitEnabled,
}) => {
    const {
        name,
        dob,
        employmentStatus,
        income,
        houseNumber,
        postCode,
    } = formDetails;
    return (
        <form className={`${className} details-form`}>
            <div>
                <label htmlFor="name">
                  Name
                    <input id="name" type="text" name="name" value={name} placeholder="Name" />
                </label>
            </div>
            <div>
                <label htmlFor="dob">
                      Date of Birth
                    <input id="dob" type="text" name="dob" value={dob} placeholder="Date of Birth" />
                </label>
            </div>
            <div>
                <label htmlFor="employment">
                      Employment Status
                    <input id="employment" type="text" name="employment" value={employmentStatus} placeholder="Employment Status" />
                </label>
            </div>
            <div>
                <label htmlFor="income">
                      Income
                    <input id="income" type="text" name="income" value={income} placeholder="Income" />
                </label>
            </div>
            <div>
                <label htmlFor="houseNo">
                      House No
                    <input id="houseNo" type="text" name="houseNo" value={houseNumber} placeholder="House No" />
                </label>
            </div>
            <div>
                <label htmlFor="postcode">
                      Postcode
                    <input id="postcode" type="text" name="postcode" value={postCode} placeholder="Postcode" />
                </label>
            </div>
            <button disabled={submitEnabled === false} type="submit" onClick={submitForm}>Show me eligible cards</button>
        </form>
    );
};

DetailsForm.propTypes = {
    className: PropTypes.string.isRequired,
    formDetails: PropTypes.shape({
        name: PropTypes.oneOfType([
            PropTypes.string,
            PropTypes.number,
        ]),
        dob: PropTypes.oneOfType([
            PropTypes.string,
            PropTypes.number,
        ]),
        employmentStatus: PropTypes.oneOfType([
            PropTypes.string,
            PropTypes.number,
        ]),
        income: PropTypes.oneOfType([
            PropTypes.string,
            PropTypes.number,
        ]),
        houseNumber: PropTypes.oneOfType([
            PropTypes.string,
            PropTypes.number,
        ]),
        postCode: PropTypes.oneOfType([
            PropTypes.string,
            PropTypes.number,
        ]),
    }).isRequired,
    submitForm: PropTypes.func.isRequired,
    submitEnabled: PropTypes.bool.isRequired,
};

export default DetailsForm;
