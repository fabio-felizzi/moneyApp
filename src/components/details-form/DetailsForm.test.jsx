import React from 'react';
import { shallow } from 'enzyme';
import DetailsForm from './DetailsForm';

const componentProps = {
    className: 'test-form',
    formDetails: {
        name: 'test-name',
        dob: '12/12/12',
        employmentStatus: 'full-time',
        income: 50000,
        houseNumber: 123,
        postCode: 'ABC 123',
    },
    submitForm: () => {},
    submitEnabled: false,
};

const {
    name,
    dob,
    employmentStatus,
    income,
    houseNumber,
    postCode,
} = componentProps.formDetails;

const componentShallow = shallow(<DetailsForm {...componentProps} />);

describe('DetailsForm component', () => {
    describe('Component Shapshot', () => {
        test('should match stored snapshot', () => {
            expect(componentShallow).toMatchSnapshot();
        });
    });

    describe('Component Render', () => {
        test('should render the name prop', () => {
            expect(componentShallow.find('#name').props().value).toBe(name);
        });

        test('should render the dob prop', () => {
            expect(componentShallow.find('#dob').props().value).toBe(dob);
        });

        test('should render the employmentStatus prop', () => {
            expect(componentShallow.find('#employment').props().value).toBe(employmentStatus);
        });

        test('should render the income prop', () => {
            expect(componentShallow.find('#income').props().value).toBe(income);
        });

        test('should render the houseNumber prop', () => {
            expect(componentShallow.find('#houseNo').props().value).toBe(houseNumber);
        });

        test('should render the postCode prop', () => {
            expect(componentShallow.find('#postcode').props().value).toBe(postCode);
        });
    });
});
