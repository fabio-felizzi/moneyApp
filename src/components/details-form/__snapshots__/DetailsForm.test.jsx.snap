// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsForm component Component Shapshot should match stored snapshot 1`] = `
<form
  className="test-form details-form"
>
  <div>
    <label
      htmlFor="name"
    >
      Name
      <input
        id="name"
        name="name"
        placeholder="Name"
        type="text"
        value="test-name"
      />
    </label>
  </div>
  <div>
    <label
      htmlFor="dob"
    >
      Date of Birth
      <input
        id="dob"
        name="dob"
        placeholder="Date of Birth"
        type="text"
        value="12/12/12"
      />
    </label>
  </div>
  <div>
    <label
      htmlFor="employment"
    >
      Employment Status
      <input
        id="employment"
        name="employment"
        placeholder="Employment Status"
        type="text"
        value="full-time"
      />
    </label>
  </div>
  <div>
    <label
      htmlFor="income"
    >
      Income
      <input
        id="income"
        name="income"
        placeholder="Income"
        type="text"
        value={50000}
      />
    </label>
  </div>
  <div>
    <label
      htmlFor="houseNo"
    >
      House No
      <input
        id="houseNo"
        name="houseNo"
        placeholder="House No"
        type="text"
        value={123}
      />
    </label>
  </div>
  <div>
    <label
      htmlFor="postcode"
    >
      Postcode
      <input
        id="postcode"
        name="postcode"
        placeholder="Postcode"
        type="text"
        value="ABC 123"
      />
    </label>
  </div>
  <button
    disabled={true}
    onClick={[Function]}
    type="submit"
  >
    Show me eligible cards
  </button>
</form>
`;
